lib/IAMClient.js

```diff
@@ -120,21 +120,26 @@ class VaultClient {
             name: accountName,
             emailAddress: options.email,
         };
-        if (options.quota !== undefined && options.quota !== null) {
-            const conv = Number.isNaN(Number.parseInt(options.quota, 10));
-            assert(typeof options.quota === 'number'
+        const { quota, externalAccountId, cdTenantId } = options;
+        if (quota) {
+            const conv = Number.isNaN(Number.parseInt(quota, 10));
+            assert(typeof quota === 'number'
                 && !conv
-                && options.quota >= 0, 'Quota must be a non-negative number');
-            data.quotaMax = options.quota;
+                && quota >= 0, 'Quota must be a non-negative number');
+            data.quotaMax = quota;
         }
-        if (options.externalAccountId !== undefined
-            && options.externalAccountId !== null) {
-            const conv = Number.isNaN(Number
-                .parseInt(options.externalAccountId, 10));
-            assert(typeof options.externalAccountId === 'string'
-                && !conv && regexAccountId.test(options.externalAccountId),
+        if (externalAccountId) {
+            const conv = Number.isNaN(Number.parseInt(externalAccountId, 10));
+            assert(typeof externalAccountId === 'string'
+                && !conv
+                && regexAccountId.test(externalAccountId),
             'invalid account id supplied');
-            data.externalAccountId = options.externalAccountId;
+            data.externalAccountId = externalAccountId;
+        }
+        if (cdTenantId) {
+            assert(typeof cdTenantId === 'string',
```

Comment:
```suggestion
            assert((cdTenantId && Array.isArray(cdTenantId)) || !cdTenantId,
```

Reply:
Same here. It's a single ID. Why should it be a list for the user?

Reply:
changed in squash commit.